From 1af9e901d51673f652d36848b26c9b8ba49c53c3 Mon Sep 17 00:00:00 2001
From: raosrini <srinidhi.rao@intel.com>
Date: Tue, 3 Mar 2015 11:51:41 +0530
Subject: [PATCH] EM/FG: Copy V-OCV instead of V-BATT when device resumes

When the device goes to suspend mode, the average current
consumption will be very low and battery will be in relaxed
state. Hence when device resumes, it will be good to read
VOCV, then do a lookup of SoC based on VOCV and adjust any
error accumulated in estimated SoC.
Hence reading V-OCV and passing it to FG Algo Library
instead of VBAT when the device resumes from suspend mode.

Change-Id: I3d0296ff019262eacb7d15307ba477d651266702
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1116
Reviewed-on: https://android.intel.com/#/c/337333
Signed-off-by: raosrini <srinidhi.rao@intel.com>
---
 drivers/power/intel_fuel_gauge.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/power/intel_fuel_gauge.c b/drivers/power/intel_fuel_gauge.c
index b725fe2..34a1de5 100644
--- a/drivers/power/intel_fuel_gauge.c
+++ b/drivers/power/intel_fuel_gauge.c
@@ -235,6 +235,15 @@ static void intel_fg_worker(struct work_struct *work)
 	if (ret)
 		dev_err(fg_info->dev, "Error while getting OCV\n");
 
+	/*
+	 * If the device was in suspend mode, then send V-OCV to
+	 * FG Algo library instead of VBAT, so that the FG Algo
+	 * lib can apply some error corrections based on V-OCV
+	 * lookup table.
+	 */
+	if (info_ptr->wake_ui.wake_enable)
+		ip.vbatt = ip.vocv;
+
 	ret = fg_info->input->get_i_avg(&ip.iavg);
 	if (ret)
 		dev_err(fg_info->dev, "Error while getting Current Average\n");
-- 
1.9.1

